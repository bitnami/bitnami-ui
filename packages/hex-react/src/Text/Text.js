import React from 'react';
import PropTypes from 'prop-types';
import cs from 'classnames';
import { Typography, cleanProps } from '../helpers';

// Outside the Box!
const Text = props => {
  const { className, children, type } = props;

  // Setup the proper CSS classes and clean the props
  const helperClasses = cs(className, Typography.propToClass(props));
  const remainingProps = cleanProps(props, Typography);

  return React.createElement(
    type,
    {
      className: helperClasses,
      ...remainingProps,
    },
    children,
  );
};

Text.propTypes = {
  /**
   * Set the HTML tag that will be rendered
   */
  type: PropTypes.oneOf(['p', 'span', 'div', 'test']),
  /**
   * Base CSS class for the component
   */
  className: PropTypes.string,
  /**
   * Any element that will be wrapped by the Box
   */
  children: PropTypes.oneOfType([PropTypes.arrayOf(PropTypes.node), PropTypes.node]),
  /* eslint-disable react/no-unused-prop-types */
  /**
   * Set the font size
   */
  fontSize: PropTypes.oneOf([
    'tiny',
    'small',
    'normal',
    'big',
    'bigger',
    'enormous',
    'giant',
    'huge',
  ]),
  /**
   * Set the color of the text
   */
  color: PropTypes.oneOf([
    'brand',
    'accent',
    'accent-light',
    'iron',
    'action',
    'white',
    'light-blue',
  ]),
  /**
   * Set the font weight
   */
  weight: PropTypes.oneOf(['light', 'regular', 'bold']),
  /**
   * Apply a text transformation to the text
   */
  textTransform: PropTypes.oneOf(['uppercase', 'capitalize']),
  /**
   * Set the text alignment
   */
  textAlign: PropTypes.oneOf(['left', 'center', 'right']),
  /* eslint-enable react/no-unused-prop-types */
};

Text.defaultProps = {
  type: 'p',
  className: '',
  children: undefined,
  fontSize: 'normal',
  color: undefined,
  weight: 'regular',
  textTransform: undefined,
  textAlign: 'left',
};

export default Text;
