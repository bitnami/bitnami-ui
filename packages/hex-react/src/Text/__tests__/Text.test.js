import React from 'react';
import { shallow } from 'enzyme';

import Text from '../Text';

describe('Text', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<Text>Test</Text>);
    expect(wrapper).toMatchSnapshot();
  });

  it('apply correctly the helpers', () => {
    const wrapper = shallow(
      <Text fontSize="small" textAlign="center">
        Test
      </Text>,
    );
    expect(wrapper).toMatchSnapshot();
  });
});
