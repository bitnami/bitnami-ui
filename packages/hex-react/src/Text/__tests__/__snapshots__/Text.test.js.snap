// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text apply correctly the helpers 1`] = `
<p
  className="type-small type-regular text-c"
  type="p"
>
  Test
</p>
`;

exports[`Text renders correctly 1`] = `
<p
  className="type-normal type-regular text-l"
  type="p"
>
  Test
</p>
`;
